utils/src/main/java/software/amazon/awssdk/utils/OptionalUtils.java

```diff
@@ -51,4 +51,17 @@ private OptionalUtils() {}
 
         return Optional.empty();
     }
+
+    public static <T> Optional<T> firstPresent(Optional<T> firstValue, Supplier<T> fallbackValue) {
+        if (firstValue.isPresent()) {
+            return firstValue;
+        }
+
+        T value = fallbackValue.get();
```

Comment:
If `fallbackValue` is empty `get()` will throw an exception here. I think we can just return `fallbackValue` here

Reply:
Oops nm, it's a `Supplier`, not an `Optional`. In that case we can just change this to `Optional.ofNullable(value)`